File: src/main/java/com/github/chMatvey/web/request/DroneRegisterRequest.java
package com.github.chMatvey.web.request;

import com.github.chMatvey.model.DroneModel;
import jakarta.validation.constraints.*;
import lombok.Builder;

@Builder
public record DroneRegisterRequest(
        @NotBlank
        @Size(max = 100)
        String serialNumber,

        @NotNull
        DroneModel model,

        @Max(500)
        @Positive
        @NotNull
        int weightLimit
) {
}

File: src/main/kotlin/com/github/chMatvey/routes/request/DroneRegisterRequest.kt
package com.github.chMatvey.routes.request

import com.github.chMatvey.models.DroneModel
import kotlinx.serialization.Serializable

@Serializable
data class DroneRegisterRequest(
    val serialNumber: String,
    val model: DroneModel,
    val weightLimit: Int,
)
